File: cs50/pset2/caesar.c
#include <stdio.h>
#include <cs50.h>
#include <ctype.h>
#include <string.h>

int main(int argc, string argv[]){
    if(argc != 2){
        printf("Error: please enter a cipher key!\n");
        return 1;
    }
    
    else {
        int key = atoi(argv[1])%26;
        printf("plaintext:\n");
        string plaintext = get_string();
        printf("ciphertext:\n");
        for(int i = 0; i<strlen(plaintext); i++){
            if(isalpha(plaintext[i])){
                if(isupper(plaintext[i])){
                    char uppercaseletter = ((((plaintext[i]-65)+key)%26)+65);
                    printf("%c", uppercaseletter);
                } else if (islower(plaintext[i])){
                    char lowercaseletter = ((((plaintext[i]-97)+key)%26)+97);
                    printf("%c", lowercaseletter);
                }
            }
            else {
                if(isdigit(plaintext[i])){
                    printf("%c", plaintext[i]);
                } else{
                    printf(" ");
                }
            }
            
        }
    }
    printf("\n");
}

File: cs50/pset6/caesar.py
import cs50
import sys

# get the cipher key from command line, check for correct num args, and convert to int
if len(sys.argv) != 2:
    print("Please enter your cipher key")

key = int(sys.argv[1])

print("plaintext:")
plaintext = cs50.get_string()
print("ciphertext:")
for i in range(len(plaintext)):
    # Check to make sure we have an alphabetic character
    if plaintext[i].isalpha() == True:
        # Determine whether the character is upper or lowercase
        if plaintext[i].isupper() == True:
            upperLetter = chr(((((ord(plaintext[i]))-65)+key)%26)+65)
            print(upperLetter, end="")
        elif plaintext[i].islower() == True:
            lowerLetter = chr(((((ord(plaintext[i]))-97)+key)%26)+97)
            print(lowerLetter, end="")
    # If the character is a digit, print the ASCII representation
    elif plaintext[i].isdigit() == True:
        print(plaintext[i], end="")
    # If we have any other character, print a space
    else:
        print(" ", end="")
# Print a new line
print()

